Installing “font-viewer@0.3.0” failed (tried 3x)

Installing “font-viewer@0.3.0” failed.Hide output…

> freetype2@0.2.3 install D:\TEMP\TMP\apm-install-dir-1151027-3976-hfpapg\node_modules\font-viewer\node_modules\freetype2
> node-gyp rebuild

D:\TEMP\TMP\apm-install-dir-1151027-3976-hfpapg\node_modules\font-viewer\node_modules\freetype2>if not defined npm_config_node_gyp (node "C:\Users\Anandajoti\AppData\Local\atom\app-1.3.0-beta6\resources\app\apm\node_modules\npm\bin\node-gyp-bin\....\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node  rebuild ) 

gypnpm ERR! Windows_NT 6.2.9200
npm ERR! argv "C:\Users\Anandajoti\AppData\Local\atom\app-1.3.0-beta6\resources\app\apm\bin\node.exe" "C:\Users\Anandajoti\AppData\Local\atom\app-1.3.0-beta6\resources\app\apm\node_modules\npm\bin\npm-cli.js" "--globalconfig" "C:\Users\Anandajoti\.atom\.apm\.apmrc" "--userconfig" "C:\Users\Anandajoti\.atom\.apmrc" "install" "D:\TEMP\TMP\d-1151027-3976-jo7q8n\package.tgz" "--target=0.34.3" "--arch=ia32"
npm ERR! node v0.10.40
npm ERR! npm  v2.13.3
npm ERR! code ELIFECYCLE

npm ERR! freetype2@0.2.3 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the freetype2@0.2.3 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the freetype2 package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls freetype2
npm ERR! There is likely additional logging output above.

Hi Eric, just tried to install again, this is the error output:
Installing “font-viewer@0.3.1” failed.Hide output…

> freetype2@0.3.0 install D:\TEMP\TMP\apm-install-dir-1151119-5388-1gesqg0\node_modules\font-viewer\node_modules\freetype2
> node-gyp rebuild


D:\TEMP\TMP\apm-install-dir-1151119-5388-1gesqg0\node_modules\font-viewer\node_modules\freetype2>if not defined npm_config_node_gyp (node "C:\Users\Anandajoti\AppData\Local\atom\app-1.4.0-beta2\resources\app\apm\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node  rebuild ) 

gypnpm ERR! Windows_NT 6.2.9200
npm ERR! argv "C:\\Users\\Anandajoti\\AppData\\Local\\atom\\app-1.4.0-beta2\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\Anandajoti\\AppData\\Local\\atom\\app-1.4.0-beta2\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "C:\\Users\\Anandajoti\\.atom\\.apm\\.apmrc" "--userconfig" "C:\\Users\\Anandajoti\\.atom\\.apmrc" "install" "D:\\TEMP\\TMP\\d-1151119-5388-pf1c59\\package.tgz" "--target=0.34.5" "--arch=ia32"
npm ERR! node v0.10.40
npm ERR! npm  v2.13.3
npm ERR! code ELIFECYCLE

npm ERR! freetype2@0.3.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the freetype2@0.3.0 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the freetype2 package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls freetype2
npm ERR! There is likely additional logging output above.

--- a/lib/font-viewer.js
+++ b/lib/font-viewer.js
@@ -75,12 +75,10 @@ export default class FontViewer {
       const chars = []
       const face = NewMemoryFace(buffer, 0);
 
-      let charcode = face.getFirstChar()
-      let num = 0
-      while (charcode !== null) {
-        num++
-        chars.push(charcode)
-        charcode = face.getNextChar(num)
+      let char = face.getFirstChar()
+      while (char && char.glyphIndex !== 0) {
+        chars.push(char.charCode)
+        char = face.getNextChar(char.charCode)
       }
 
       return callback(chars)

--- a/lib/font-viewer.js
+++ b/lib/font-viewer.js
@@ -9,7 +9,7 @@
 import path from "path"
 import fs from "fs-plus"
 import { File, CompositeDisposable } from "atom"
-import ft from "freetype2"
+import {NewMemoryFace} from "freetype2"
 
 // Editor model for a font file
 export default class FontViewer {
@@ -73,15 +73,14 @@ export default class FontViewer {
   getAvailableCharacters(callback) {
     return fs.readFile(this.getPath(), function (err, buffer) {
       const chars = []
-      const gindex = {}
-      let face = {}
-      ft.New_Memory_Face(buffer, 0, face)
-      ;({ face } = face)
-      let charcode = ft.Get_First_Char(face, gindex)
+      const face = NewMemoryFace(buffer, 0);
 
-      while (gindex.gindex !== 0) {
+      let charcode = face.getFirstChar()
+      let num = 0
+      while (charcode !== null) {
+        num++
         chars.push(charcode)
-        charcode = ft.Get_Next_Char(face, charcode, gindex)
+        charcode = face.getNextChar(num)
       }
 
       return callback(chars)

--- a/package.json
+++ b/package.json
@@ -15,7 +15,7 @@
     "atom-space-pen-views-plus": "^3.0.4",
     "fs-plus": "~3.1.1",
     "underscore-plus": "^1.7",
-    "freetype2": "1.0.1"
+    "freetype2": "1.0.3"
   },
   "activationHooks": [
     "core:loaded-shell-environment"

--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "font-viewer",
   "main": "./lib/main",
-  "version": "0.4.3",
+  "version": "0.4.2",
   "description": "Preview font files in an editor in Atom.",
   "repository": "https://github.com/ericfreese/font-viewer",
   "license": "MIT",
@@ -15,7 +15,7 @@
     "atom-space-pen-views-plus": "^3.0.4",
     "fs-plus": "~3.1.1",
     "underscore-plus": "^1.7",
-    "freetype2": "0.5.0"
+    "freetype2": "1.0.1"
   },
   "activationHooks": [
     "core:loaded-shell-environment"

